tests/relay/test_rln.py

```diff
@@ -1,24 +1,193 @@
+            assert not failed_payloads, f"Payloads failed: {failed_payloads}"
+
+    @pytest.mark.skip(reason="exceeding timeout ATM, need to check with devs")
+    @pytest.mark.timeout(600)
+    def test_valid_payloads_dynamic_at_slow_rate(self):
+        self.setup_main_rln_relay_nodes(rln_relay_dynamic="true")
+        self.subscribe_main_relay_nodes()
+        failed_payloads = []
+        for payload in SAMPLE_INPUTS:
+            logger.debug(f'Running test with payload {payload["description"]}')
+            message = self.create_message(payload=to_base64(payload["value"]))
+            try:
+                self.check_published_message_reaches_relay_peer(message)
+            except Exception as e:
+                logger.error(f'Payload {payload["description"]} failed: {str(e)}')
+                failed_payloads.append(payload["description"])
+            delay(1)
+            assert not failed_payloads, f"Payloads failed: {failed_payloads}"
+
+    @pytest.mark.skip(reason="exceeding timeout ATM, need to check with devs")
+    @pytest.mark.timeout(600)
+    def test_valid_payloads_dynamic_at_spam_rate(self):
+        self.setup_main_rln_relay_nodes(rln_relay_dynamic="true")
+        self.subscribe_main_relay_nodes()
+        previous = math.trunc(time())
+        for i, payload in enumerate(SAMPLE_INPUTS):
+            logger.debug(f'Running test with payload {payload["description"]}')
+            message = self.create_message(payload=to_base64(payload["value"]))
+            try:
+                now = math.trunc(time())
+                self.publish_message(message)
+                if i > 0 and (now - previous) == 0:
+                    raise AssertionError("Publish with RLN enabled at spam rate worked!!!")
+                else:
+                    previous = now
+            except Exception as e:
+                assert "RLN validation failed" in str(e)
+
+    def test_valid_payloads_n1_with_rln_n2_without_rln_at_spam_rate(self):
+        self.setup_first_rln_relay_node()
+        self.setup_second_relay_node()
+        self.subscribe_main_relay_nodes()
+        previous = math.trunc(time())
+        for i, payload in enumerate(SAMPLE_INPUTS):
+            logger.debug(f'Running test with payload {payload["description"]}')
+            message = self.create_message(payload=to_base64(payload["value"]))
+            try:
+                now = math.trunc(time())
+                self.publish_message(message)
+                if i > 0 and (now - previous) == 0:
+                    raise AssertionError("Publish with RLN enabled at spam rate worked!!!")
+                else:
+                    previous = now
+            except Exception as e:
+                assert "RLN validation failed" in str(e)
 
-        if RLN_CREDENTIALS is None:
-            pytest.skip("RLN_CREDENTIALS not set, skipping test")
+    @pytest.mark.skip(reason="Peer NODE_2:harbor.status.im/wakuorg/nwaku:latest couldn't find any messages")
```

Comment:
Checking with devs wether some level of compatibility is possible. https://discord.com/channels/1110799176264056863/1111541184490393691/1234360260580474932

Reply:
I also think it is somewhat valid use case, although it is not supported. If I understand well, RLN is protecting operator rather than community chatting over a certain pubsub topic. https://github.com/waku-org/waku-interop-tests/pull/30/commits/ee2e05679e5aff0dc2c8f43674883d257bb92a47